### PR TITLE
Xrddev 1702 amending csp sts headers for CS & SS Admin UI:s

### PR DESCRIPTION
**Pull request by teemukin65**

### CODE REVIEW

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java [hunk 1]
```diff
@@ -50,6 +50,14 @@ protected void configure(HttpSecurity http) throws Exception {
                         "/js/**",
                         "/fonts/**")
                 .and()
+            .headers()
+                    .contentSecurityPolicy(
```
Comment (jansu76) [inline src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java:54]: nitpick: indentation seems off, compared to other blocks

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java [hunk 2]
```diff
@@ -50,6 +50,14 @@ protected void configure(HttpSecurity http) throws Exception {
...
+                            "default-src 'self' 'unsafe-inline'  ;"
+                            + "script-src 'self' 'unsafe-inline' 'unsafe-eval' ;"
```
Comment (jansu76) [inline src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java:56]: Maybe we don't need all unsafe-xxx definitions? Implementation would in principle be a bit more secure if we can avoid some. I will put more in the Jira ticket.

**Review by jansu76 — CHANGES_REQUESTED** [review]

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java (hunk 2 above)
Comment (teemukin65) [reply src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java]: Fix proposal: abf32ba

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java (hunk 1 above)
Comment (teemukin65) [reply src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java:54]: Another take on the indentation: abf32ba

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java [hunk 3]
```diff
@@ -90,6 +89,10 @@ protected void configure(HttpSecurity http) throws Exception {
                 .ignoringAntMatchers(LOGIN_URL)
                 .csrfTokenRepository(new CookieAndSessionCsrfTokenRepository())
                 .and()
+            .headers()
+                    .contentSecurityPolicy("default-src 'self' 'unsafe-inline'")
+                    .and()
+                .and()
```
[the remark this answers is not in the document]
Comment (teemukin65) [reply src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java:95]: This approach (I think) taken in : abf32ba

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java [hunk 4]
```diff
@@ -1,27 +1,27 @@
-/**
- * The MIT License
```
[the remark this answers is not in the document]
Comment (teemukin65) [reply src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java:2]: Now licenses redone with licenseFormat - 1ba491b

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jansu76) [reply src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java:97]: Still some minor issues maybe, now it looks like this

```
            .headers()
                    .contentSecurityPolicy("default-src 'self' 'unsafe-inline'")
                    .and()
                .and()
```

and e.g. StaticAssetsWebSecurityConfig has

```
        http
                .requestMatchers()
                .antMatchers("/favicon.ico",
```

differing from FormLoginWebSecurityConfigurer's

```
        http
            .authorizeRequests()
                .antMatchers("/error").permitAll()

```
but it's really not super important issue.

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java (hunk 2 above)
Comment (teemukin65) [reply src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/StaticAssetsWebSecurityConfig.java]: Updated set of CSP -headers are added now  both to SS (8db52e5) and (at last) to CS (8b4f594).
Also STS header for CS has been added now (bc80ba7).

File: src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (teemukin65) [reply src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java]: hopefully resolve by commit 1204724

[the remark this answers is not in the document]
Comment (jansu76) [reply src/common-rest-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/FormLoginWebSecurityConfigurerAdapter.java]: Some 8 char indentations still, but IntelliJ seems to have a will of it's own, and as said not a super important issue

**Pull request merged**